lib/Relation.php

```diff
@@ -457,6 +516,7 @@ public function all(): Relation
      */
     public function count(): int
     {
+        unset($this->options['select']);
```

Comment:
Bug: I would expect

$relation = Author::where('id <> 4')->select('name');
$count = $relation->count; // returns 3
$queries = $relation->find; // Should return a Model[] but the old select() statement is overwritten

Instead of `unset($this->options['select']);`, cache it first `$oldSelect = $this->options['select'];` and then restore it after `$res = $this->count();` with `$this->options['select'] = $oldSelect;`